### PR TITLE
Add build steps for API tests

### PR DESCRIPTION
**Pull request by jrabbott**
### Context
No backlog reference

### Change proposed in this pull request
Adds build steps for API tests

### Guidance to review 
Review pipeline artifacts to ensure successfully published

### Checklist
- [x] ~~Work items have been linked [(use AB#)](https://learn.microsoft.com/en-us/azure/devops/boards/github/link-to-from-github?view=azure-devops#use-ab-to-link-from-github-to-azure-boards-work-items)~~
- [x] Your code builds clean without any errors or warnings
- [x] You have run all unit/integration tests and they pass
- [x] Your branch has been rebased onto main
- [x] You have tested by running locally

### CODE REVIEW

File: platform/pipelines/build.yaml [hunk 1]
```diff
@@ -68,6 +68,11 @@ stages:
         steps:
           - template: steps-platform-run-unit-tests.yaml
 
+      - job: BuildApiTests
+        displayName: 'Build platform apt tests'
```
Comment (WolfyUK) [inline platform/pipelines/build.yaml:72]: Mini typo ;-)

File: platform/pipelines/steps-platform-build-api-tests.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        nuget | "$(Agent.OS)-api-tests"
+      path: $(APITESTS_NUGET_CACHE)
+      cacheHitVar: APITESTS_CACHE_RESTORED
+
+  - task: DotNetCoreCLI@2
+    displayName: 'Restore nuget packages'
+    condition: ne(variables.APITESTS_CACHE_RESTORED, 'true')
+    inputs:
+      command: restore
+      projects: 'platform/tests/EducationBenchmarking.Platform.ApiTests/*EducationBenchmarking.Platform.ApiTests.csproj'
```
Comment (WolfyUK) [inline platform/pipelines/steps-platform-build-api-tests.yaml:21]: Is that an intentional asterisk here and below?

Comment (jrabbott) [reply platform/pipelines/steps-platform-build-api-tests.yaml:21]: Nope, not intentional - good spot.

File: platform/pipelines/build.yaml (hunk 1 above)
Comment (jrabbott) [reply platform/pipelines/build.yaml]: Push change

File: platform/pipelines/steps-platform-build-api-tests.yaml (hunk 2 above)
Comment (jrabbott) [reply platform/pipelines/steps-platform-build-api-tests.yaml]: Push change

**Review by WolfyUK — APPROVED** [review]

**Pull request merged**